### PR TITLE
rust: publish a crate on release

### LINKED ISSUE #22: Publish on crates.io

### PR DESCRIPTION
**Pull request by Tristramg**
closes #22

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  release:
+    types: [published]
+
+jobs:
+  publish:
+    name: Publish on crates.io
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions-rs/toolchain@v1
```
Comment (flomonster) [inline .github/workflows/publish.yaml:11]: `action-rs` is archive and unmaintained. There is no widely used and maintained github action for rust. The best practice is to manually install rustup and run build your crate.

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        with:
+            toolchain: stable
+            override: true
+      - uses: katyo/publish-crates@v2
+        with:
+            registry-token: ${{ secrets.CARGO_REGISTRY_TOKEN }}
```
Comment (flomonster) [inline .github/workflows/publish.yaml:17]: Missing line feed.

> [!NOTE]
> We have a CI script to check this in OSRD.

**Review by woshilapin — APPROVED** [review]

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (Tristramg) [reply .github/workflows/publish.yaml]: actually everything seems included by default now (I suppose that’s why the cation deprecated)

**Comment by Tristramg** [comment]
Thank you! @flomonster so apparently we need neither custom rust action nor installing rustup&co by hand now.

I made separate commits to show the changes since your review, but i plan squashing them at merge

**Review by flomonster — APPROVED** [review]
LGTM

**Pull request merged**